modules/fedimint-gateway-v1/src/lib.rs

```diff
@@ -862,3 +870,52 @@ impl TryFrom<PaymentData> for SwapParameters {
         })
     }
 }
+
+#[async_trait]
+pub trait LightningV1Manager: Debug + Send + Sync {
+    // Checks if the invoice route hint last hop has source node id matching this
+    // gateways node pubkey and if the short channel id matches one assigned by
+    // this gateway to a connected federation. In this case, the gateway can
+    // avoid paying the invoice over the lightning network and instead perform a
+    // direct swap between the two federations.
+    async fn check_swap_to_federation(
+        &self,
+        payment_data: PaymentData,
+    ) -> Option<Spanned<ClientHandleArc>>;
+
+    /// Verifies that the supplied `preimage_auth` is the same as the
+    /// `preimage_auth` that initiated the payment. If it is not, then this
+    /// will return an error because this client is not authorized to receive
+    /// the preimage.
+    async fn verify_preimage_authentication(
+        &self,
+        payment_hash: sha256::Hash,
+        preimage_auth: sha256::Hash,
+        contract: OutgoingContractAccount,
+    ) -> Result<(), OutgoingPaymentError>;
+
+    /// Retrieves the federation's routing fees from the federation's config.
+    async fn get_routing_fees(&self, federation_id: FederationId) -> Option<RoutingFees>;
+
+    /// Retrieve a client given a federation ID, used for swapping ecash between
+    /// federations.
+    async fn get_client(&self, federation_id: &FederationId) -> Option<Spanned<ClientHandleArc>>;
+
+    /// Pay a Lightning invoice using the gateway's lightning node.
+    async fn pay(
+        &self,
+        payment_data: PaymentData,
+        max_delay: u64,
+        max_fee: Amount,
+    ) -> Result<PayInvoiceResponse, LightningRpcError>;
+
+    /// Verify that the lightning node supports private payments if a pruned
+    /// invoice is supplied.
+    async fn check_pruned_invoice(&self, payment_data: PaymentData) -> anyhow::Result<()>;
```

Comment:
If you use "verify" in the docstring, then the method probably should start with "verify" as well. Best names usually match very closely the fist line in docstring.